packages/grzctl/src/grzctl/commands/db.py

```diff
@@ -275,31 +276,35 @@ class SignatureStatus(enum.StrEnum):
     ERROR = "Error"
     UNKNOWN = "Unknown"
 
-    def rich_display(self) -> str:
+    def rich_display(self, comment: str | None) -> str:
         """Displays the signature status in rich format."""
         match self:
             case "Verified":
-                return "[green]Verified[/green]"
+                return "[green]Verified[/green]" if comment is None else f"[green]Verified ({comment})[/green]"
             case "Failed":
                 return "[red]Failed[/red]"
             case "Error":
                 return "[red]Error[/red]"
             case "Unknown" | _:
-                return "[yellow]Unknown[/yellow]"
+                return "[yellow]Unknown Key[/yellow]"
 
 
-def _verify_signature(author_public_key, verifiable_log: VerifiableLog) -> SignatureStatus:
+def _verify_signature(
+    public_keys: dict[str, Ed25519PublicKey], verifiable_log: VerifiableLog
+) -> tuple[SignatureStatus, str | None]:
     signature_status = SignatureStatus.UNKNOWN
-    if author_public_key:
+    verifying_key_comment = None
+    for comment, public_key in public_keys.items():
         try:
-            if verifiable_log.verify(author_public_key):
+            if verifiable_log.verify(public_key):
                 signature_status = SignatureStatus.VERIFIED
-            else:
-                signature_status = SignatureStatus.FAILED
+                verifying_key_comment = comment
+                # stop trying after first verification success
+                break
         except Exception as e:
             signature_status = SignatureStatus.ERROR
             log.error(e)
-    return signature_status
+    return signature_status, verifying_key_comment
```

Comment:
Do we really want to log every attempt that fails, now that we just iterate all known public keys?
And how do we now differentiate between: "no known keys to verify this" vs "something went wrong during verification (with any key)".